contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -68,7 +68,15 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 		logger.Infof("Received message from %q channel", channel.String())
 		// publish to Natss
 		ch := getSubject(channel)
-		if err := stanutil.Publish(s.natssConn, ch, &m.Payload, logger); err != nil {
+		var message []byte
+		for k, v := range m.Headers {
```

Comment:
I think this is all parsed as part of the payload, line 158. We need to make sure that the message's headers are parsed as message headers down there, and not part of the payload.

Reply:
done